homeassistant/components/yamaha_musiccast/media_player.py

```diff
@@ -166,6 +166,27 @@ def state(self):
             return STATE_PLAYING
         return STATE_OFF
 
+    @property
+    def source_mapping(self):
+        """Return a mapping of the actual source names to their labels configured in the MusicCast App."""
+        ret = {}
+        for inp in self.coordinator.data.zones[self._zone_id].input_list:
+            label = self.coordinator.data.input_names.get(inp, "")
+            if inp != label and (
+                label in self.coordinator.data.zones[self._zone_id].input_list
```

Comment:
Good. This checks handles any possible clash between the user entered label and the original input name. Overall this logic looks solid.